db/controllers/topics.controller.js

```diff
@@ -39,23 +40,34 @@ exports.getArticleById = (request, response, next) => {
 };
 
 exports.getArticles = (request, response, next) => {
-  countCommentTotals().then((idCounts) => {
-    retrieveAllArticles(idCounts)
-      .then((articles) => {
-        const countsToMap = {}
-        idCounts.forEach((articleCount) => {
-          countsToMap[articleCount.article_id] = Number(articleCount.total)
+  countCommentTotals()
+    .then((idCounts) => {
+      retrieveAllArticles(idCounts)
+        .then((articles) => {
+          const countsToMap = {};
+          idCounts.forEach((articleCount) => {
+            countsToMap[articleCount.article_id] = Number(articleCount.total);
+          });
+          articles.forEach((article) => {
+            article.comment_count = countsToMap[article.article_id] || 0;
+          });
```

Comment:
You should be getting the comment_count from within the query in the model - have a look at COUNT.